### PR TITLE
Create presubmit job for build cluster validation

### PR DESCRIPTION
**Pull request by smg247**

### CODE REVIEW

**Comment by smg247** [comment]
The only failing test is one that has been removed as it is made redundant by the new `validate-ci-build-clusters` presubmit.

/cc @openshift/test-platform 
/label tide/merge-method-squash

File: hack/validate-ci-build-clusters.sh
Comment (hongkailiu) [inline hack/validate-ci-build-clusters.sh:17]: nit: we could work on the current dir, and `git diff` on it.
See an example: https://github.com/openshift/release/blob/master/ci-operator/jobs/openshift/release/openshift-release-master-presubmits.yaml#L1113-L1126

We need to add git into the image if you want to go that way.

**Comment by hongkailiu** [comment]
/lgtm

/hold
in case you want to handle the nit or you are ready to watch the deployment
The green rehearsal is a good sign.

To unblock the (deleted) failed presubmit, you can override it.

File: hack/validate-ci-build-clusters.sh
Comment (stevekuznetsov) [reply hack/validate-ci-build-clusters.sh:17]: Probably fine as is

**Review by stevekuznetsov — COMMENTED** [review]
Could we get a handy `make` target to update the state of the world so that when a user fails this they know what to run to fix it? See the comments in the other presubmits for an example/

File: hack/validate-ci-build-clusters.sh
Comment (bbguimaraes) [reply hack/validate-ci-build-clusters.sh:17]: Heh, I was going to suggest the same in the original PR, but after thinking about it a bit, we probably do this to make it easier to execute these tests in a local clone without disturbing the working directory.

**Comment by smg247** [comment]
> Could we get a handy `make` target to update the state of the world so that when a user fails this they know what to run to fix it? See the comments in the other presubmits for an example/

Thats a good idea! I added it.

**Comment by smg247** [comment]
@hongkailiu, how can I override the defunct job?

**Review by stevekuznetsov — APPROVED** [review]

**Pull request merged**
